pkg/cloudprovider/providers/gce/kms.go

```diff
@@ -129,19 +134,23 @@ func (gce *GCECloud) getGCPCloudKMSService(config io.Reader) (envelope.Service,
 	return service, nil
 }
 
-// Decrypt decrypts a base64 representation of encrypted bytes.
+// Decrypt decrypts a base64 representation of (key-version-used-to-encrypt + encrypted bytes).
 func (t *gkmsService) Decrypt(data string) ([]byte, error) {
+	dataChunks := strings.SplitN(data, ":", 2)
+	if len(dataChunks) != 2 {
+		return []byte{}, fmt.Errorf("invalid data encountered for decryption: %s. Missing key version", data)
```

Comment:
Why aren't you returning nil instead of an empty byte array?

Reply:
Fixed.